### PR TITLE
Submitting Answers

### PR DESCRIPTION
**Pull request by tgaeta**

### CODE REVIEW

File: HalfWayChallenge.md [hunk 1]
```diff
@@ -1,90 +1,91 @@
...
+It's a definition. You're defining something; a single object/concept. Classes also begin with a capital letter (Dog Cat Bird).
 
 
+2. What is an attribute?<br> -
+Attributes are what gives the class structure. I'm a Person (notice the capital P). I have a name, age, and sex. These few attributes are what make me a Person.
 
-2. What is an attribute?
 
+3. What is an object?<br> -
+This is sort of a trick question. It's like you said in week two. Depending on who you talk to, it's either an incredibly complex answer, or a very simple one. From my understanding, an object is anything that can be manipulated.
```
Comment (brianburridge) [inline HalfWayChallenge.md]: I wouldn't say it's a trick question, or that it really needs to be complex. Simply put, it's an instance of a class, with a combination of variables, functions, and data structures

File: HalfWayChallenge.md [hunk 2]
```diff
@@ -1,90 +1,91 @@
...
 
 
-3. What is an object?
+4. What type of object is a PORO?<br> -
+Well, first, it's meaning... Plain Old Ruby Object. I guess any object that doesn't use ActiveRecord in a Rails app. Just some plain old Ruby.
 
 
+5. What is a Rails model?<br> -
+A Rails model houses your classes. These models can be parents or children. It uses ActiveRecord to convert Ruby passed from the controller into SQL when communicating with the database.
```
Comment (brianburridge) [inline HalfWayChallenge.md]: Note: A Rails model is a class, it doesn't contain, or "house" a class.

File: HalfWayChallenge.md [hunk 3]
```diff
@@ -1,90 +1,91 @@
...
+Gems are libraries and/or packages of Ruby code that extend and add functionality to your apps.
 
 
-12. What does the pattern MVC mean?
+17. In what part of a Rails application do we tell it how to handle an HTTP request to a specific path, like: `http://localhost:3000/movies`?<br> -
+In our routes.rb file located in the config folder.
 
 
+18. In what part of a Rails application do we hold data provided by the user, in order to use it, save it to the database, or retrieve it from the database?<br> -
+This is a tricky question. Initially I wanted to say the database, but it's at the model where CRUD (Create, Read, Update, Delete) is occurring.
```
Comment (brianburridge) [inline HalfWayChallenge.md]: Yes, model is what I was getting at here. :+1:

File: HalfWayChallenge.md [hunk 4]
```diff
@@ -94,93 +95,92 @@
...
+39. What gem can we use to provide a fast, yet custom and robust, administrative section for our apps?<br> -
+`gem 'activeadmin', github: 'activeadmin'`
 
 
-41. If a User has many ParkingTickets, and the `user` variable points to a User object, what code would you use to get all the ParkingTickets?
+40. What is does _anopisthographic_ mean?<br> -
+I still haven't found an answer, but I'm not giving up. I'll get back to you on this.
 
 
+41. If a User has many ParkingTickets, and the `user` variable points to a User object, what code would you use to get all the ParkingTickets?<br> - `user.parking_tickets.all` should do the trick.
```
Comment (brianburridge) [inline HalfWayChallenge.md]: The `.all` isn't necessary. Just `user.parking_tickets`

File: HalfWayChallenge.md [hunk 5]
```diff
@@ -94,93 +95,92 @@
...
-44. If we get this message in our log after attempting to save an object, what does it mean?
+44. If we get this message in our log after attempting to save an object, what does it mean? - You forgot to pass the first_name object to the params permit in the private method section of the controller you're calling on.
 
 
         Unpermitted parameters: first_name
 
 
-45. What one question do you wish I would have asked?
+45. What one question do you wish I would have asked?<br> -
+Was Bin Ladin's body really dropped in the North Arabian Sea?
```
Comment (brianburridge) [inline HalfWayChallenge.md]: Yeah, but if I had asked that, you couldn't have answered it anyway. :wink:

**Comment by brianburridge** [comment]
:+1:

**Comment by tgaeta** [comment]
@brianburridge I would have... But I would have made you say your goodbyes... haha :)

**Pull request closed without merging**